packages/dds/tree/src/feature-libraries/flex-tree/lazyField.ts

```diff
@@ -139,6 +130,16 @@ export abstract class LazyField<TKind extends FlexFieldKind, TTypes extends Flex
 		super(context, schema, cursor, fieldAnchor);
 		assert(cursor.mode === CursorLocationType.Fields, 0x77b /* must be in fields mode */);
 		this.key = cursor.getFieldKey();
+		// Fields currently live as long as their parent does.
+		// For root fields, this means forever, but other cases can be cleaned up when their parent anchor is deleted.
+		if (fieldAnchor.parent !== undefined) {
+			const anchorNode =
+				context.forest.anchors.locate(fieldAnchor.parent) ??
+				fail("parent anchor node should always exist since field is under a node");
+			this.offAfterDestroy = anchorNode.on("afterDestroy", () => {
+				this[disposeSymbol]();
```

Comment:
It might be simpler to just call `offAfterdestroy()` at the end of this method (it's totally fine for an event handler to close over its own `off` function), rather than saving it as a class member. Functionally it's the same, since [dispose] is the only thing that calls [forgetAnchorSymbol].

Reply:
Sounds fine to me so long that we can reasonably expect "[dispose] is the only thing that calls [forgetAnchorSymbol]" not to change. @CraigMacomber, thoughts ?

Reply:
Actually, wait. If somebody else calls dispose, then we want to make sure this event still deregisters. So maybe it does need to be here? I'm not sure what actually triggers the anchor "afterDestroy" event. If dispose is called from elsewhere, will "afterDestroy" fire or not?

#SpaghettiEvents

Reply:
Any uncertainty in this space makes me want to keep the more conservative implementation.

Reply:
If you read the doc comments on [forgetAnchorSymbol], you will see that its call on dispose if the entity has an anchor. This type unconditionally has an anchor, so this is called on dispose and only on dispose.

Unregistering events from an anchor when the anchor is disposed due to being perminatly deleated from the tree/anchorSet (which is when afterDestroy runs) is needless as the anchor is invalid to use and will never have any more events and should have no references keeping it in memory. Thus adding it to the callback would double unregister, but only when unregistering is not needed.

Reply:
Disposing of a flex-tree (ex: when a view is disposed) does not impact the anchor set at all. Thus this afterDestroy callback does not run when disposing a context or view, only when the actual node stops existing in the anchorset.

Reply:
A flex tree stays alive as long as the node its corresponds to is not destroyed and its context still exists. The end of live of either can dispose the node or field. I added tests covering both cases.